lms/static/scripts/frontend_apps/components/test/AutoGradingConfigurator-test.js

```diff
@@ -0,0 +1,172 @@
+import { checkAccessibility } from '@hypothesis/frontend-testing';
+import { mount } from 'enzyme';
+import { act } from 'preact/test-utils';
+
+import AutoGradingConfigurator from '../AutoGradingConfigurator';
+
+describe('AutoGradingConfigurator', () => {
+  let fakeAutoGradingConfig;
+  let fakeUpdateAutoGradingConfig;
+
+  beforeEach(() => {
+    fakeAutoGradingConfig = {
+      gradingType: 'all_or_nothing',
+      activityCalculation: 'cumulative',
+      requiredAnnotations: 1,
+    };
+    fakeUpdateAutoGradingConfig = sinon.stub();
+  });
+
+  function createComponent() {
+    return mount(
+      <AutoGradingConfigurator
+        autoGradingConfig={fakeAutoGradingConfig}
+        updateAutoGradingConfig={fakeUpdateAutoGradingConfig}
+      />,
+    );
+  }
+
+  function dispatchOnChange(wrapper, selector, event) {
+    act(() => wrapper.find(selector).props().onChange(event));
+  }
+
+  [true, false].forEach(active => {
+    it('renders components if auto grading is active', () => {
+      fakeAutoGradingConfig.active = active;
+      const wrapper = createComponent();
+
+      assert.equal(wrapper.exists('RadioGroup'), active);
+    });
+
+    it('updates config when checkbox is changed', () => {
+      const wrapper = createComponent();
+
+      dispatchOnChange(wrapper, 'Checkbox', {
+        target: { checked: active },
+      });
+
+      assert.calledWith(fakeUpdateAutoGradingConfig, sinon.match({ active }));
+    });
+  });
+
+  context('when auto grading is active', () => {
+    beforeEach(() => {
+      fakeAutoGradingConfig.active = true;
+    });
+
+    ['cumulative', 'separately'].forEach(activityCalculation => {
+      it('updates config when changing activity calculation', () => {
+        const wrapper = createComponent();
+
+        dispatchOnChange(
+          wrapper,
+          '[data-testid="activity-calculation-radio-group"]',
+          activityCalculation,
+        );
+
+        assert.calledWith(
+          fakeUpdateAutoGradingConfig,
+          sinon.match({ activityCalculation }),
+        );
+      });
+
+      it('renders inputs based on activity calculation value', () => {
+        fakeAutoGradingConfig.activityCalculation = activityCalculation;
+
+        const wrapper = createComponent();
+        const inputs = wrapper.find('AnnotationsGoalInputGroup');
+        const firstInput = inputs.first();
+
+        assert.equal(
+          activityCalculation === 'separately' ? 2 : 1,
+          inputs.length,
```

Comment:
Arguments are the wrong way round here. It should always be `assert.equal(actual, expected)` - https://www.chaijs.com/api/assert/#method_equal